aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -678,14 +672,8 @@ impl AptosVM {
             txn_data.gas_unit_price,
         )?;
 
-        let storage_with_changes =
-            DeltaStateView::new(resolver, inner_function_change_set_ext.write_set());
-        let delta_write_set = inner_function_change_set_ext
-            .delta_change_set()
-            .clone()
-            .try_into_write_set(resolver)?;
-        let delta_state_view = DeltaStateView::new(&storage_with_changes, &delta_write_set);
-        let resolver = delta_state_view.as_move_resolver();
+        let storage_with_changes = DeltaStateView::new(resolver, &inner_function_change_set_ext);
+        let resolver = storage_with_changes.as_move_resolver();
```

Comment:
same - could combine in one line if we never use 'storage_with_changes'

Reply:
When I combine the two lines into one, it gives me "creates a temporary which is freed while still in use" error. So, I left it like this.